maro/cli/inspector/visualization.py

```diff
@@ -0,0 +1,656 @@
+        down_pooling_num = st.sidebar.select_slider(
+            'Epoch Sampling Ratio',
+            sample_ratio)
+        down_pooling_len = math.floor(1 / down_pooling_num)
+        down_pooling_range = generate_down_pooling_sample(epoch_num, down_pooling_len, start_epoch, end_epoch)
+        item_option_all = ['All', 'Booking Info', 'Port Info', 'shortage', 'booking', 'fulfillment', 'on_shipper',
+                        'on_consignee', 'capacity', 'full', 'empty']
+        data = read_summary_data(ports_file_path, down_pooling_range)
+        data_genera = formula_define(data)
+        if data_genera is not None:
+            data = data_genera['data_after']
+            item_option_all.append(data_genera['name'])
+        item_option = st.multiselect(
+            ' ', item_option_all, item_option_all)
+        item_option = get_CIM_item_option(item_option, item_option_all)
+        data = data[item_option]
+        generate_summary_plot(item_option, data, down_pooling_range)
+    else:
+        data = pd.read_csv(stations_file_path)
+        name_conversion = read_name_conversion(CONVER_PATH)
+        data['station name'] = list(map(lambda x: name_conversion.loc[int(x[9:])][0], data['name']))
+        df_bikes = data[['station name', 'bikes']].sort_values(by='bikes', ascending=False).head(5)
+        df_requirement = data[['station name', 'trip_requirement']].sort_values(by='trip_requirement',
+                                                                            ascending=False).head(5)
+        df_fulfillment = data[['station name', 'fulfillment']].sort_values(by='fulfillment', ascending=False).head(5)
+        df_fulfillment_ratio = data[['station name', 'fulfillment_ratio']].sort_values(by='fulfillment_ratio',
+                                                                                    ascending=False).head(5)
+        generate_by_snapshot_top_summary('station name', df_bikes, 'bikes', False)
+        generate_by_snapshot_top_summary('station name', df_requirement, 'trip_requirement', False)
+        generate_by_snapshot_top_summary('station name', df_fulfillment, 'fulfillment', False)
+        generate_by_snapshot_top_summary('station name', df_fulfillment_ratio, 'fulfillment_ratio', False)
+
+
+# generate down pooling list based on origin data and down pooling rate
+def generate_down_pooling_sample(origin_len, down_pooling_len, start_epoch, end_epoch):
+    down_pooling_range = list(range(start_epoch, end_epoch, down_pooling_len))
+    if end_epoch not in down_pooling_range:
+        down_pooling_range.append(end_epoch)
+    return down_pooling_range
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument("rootpath")
+    args = parser.parse_args()
+
+    ROOT_PATH = args.rootpath
+
+    # path to restore summary files
+    ports_file_path = os.path.join(ROOT_PATH, r'snapshot_ports_summary.csv')
+    vessels_file_path = os.path.join(ROOT_PATH, r'snapshot_vessels_summary.csv')
```

Comment:
check with lvchun, mv to https://github.com/microsoft/maro/blob/master/maro/cli/utils/params.py